index.js

```diff
@@ -66,48 +75,45 @@ class Scholar {
   async getPubAuthors(query) {
     const { authors } = await this.searchPub(query);
     if (!authors) {
-      return null;
+      return;
     }
-    return await Promise.all(authors.map(async ({ url }) => await this.getAuthorProfile(url)));
+    return Promise.all(authors.map(({ url }) => this.getAuthorProfile(url)));
   }
 
   parsePub(html) {
-    const { pub } = searchTags;
+    const { pub } = selectors;
     const $ = cheerio.load(html);
-    const $result = $(pub.result).first().html();
-    const $authors = $(pub.authors, $result);
-
-    const authors = [];
-    $($authors).each((_, $element) => {
-      const name = $($element).text();
-      const url = $($element).attr('href');
-      authors.push({ name, url });
-    });
-    const title = $(pub.title, $result).text();
+    const $publicationContainer = $(pub.container).first();
+
+    const $authors = $publicationContainer.find(pub.authors);
+    const authors = $authors.map((_, el) => {
+      const $el = $(el);
+      const name = $el.text();
+      const url = $el.attr('href');
+      return { name, url };
+    }).get();
+
+    const title = $publicationContainer.find(pub.ttitle).text();
```

Comment:
:pencil2:  ~~ttitle~~ -->  title

Reply:
Yup, ESLint is not enough, I also need a spelling checker 🤦